akka-stream/src/main/scala/akka/stream/impl/io/SslTlsCipherActor.scala

```diff
@@ -450,3 +474,23 @@ private[akka] class SslTlsCipherActor(settings: ActorMaterializerSettings, sslCo
     context.stop(self)
   }
 }
+
+/**
+ * Gives the chance to configure the SSLContext before it is going to be used.
+ * The passed in context will be already set in client mode and provided with hostInfo during initialization.
+ */
+private[akka] trait SSLEngineConfigurator {
+  def configure(engine: SSLEngine, sslContext: SSLContext): SSLEngine
+}
+
+/** Factory that creates an SSLEngine. */
+private[akka] final class DefaultSSLEngineConfigurator(config: SSLConfig, enabledProtocols: Array[String], enabledCipherSuites: Array[String])
+  extends SSLEngineConfigurator {
+  def configure(engine: SSLEngine, sslContext: SSLContext): SSLEngine = {
+    engine.setSSLParameters(sslContext.getDefaultSSLParameters)
+    engine.setEnabledProtocols(enabledProtocols)
+    engine.setEnabledCipherSuites(enabledCipherSuites)
+    engine.setUseClientMode(true)
+    engine
+  }
```

Comment:
Will think about if we can't more simply just make it an extension for example (in ssl-config-akka).